TransformationSystem/Agent/TransformationCleaningAgent.py

```diff
@@ -534,14 +534,16 @@ def __removeRequests(self, requestIDs):
     """
     rIDs = [int(long(j)) for j in requestIDs if long(j)]
     for reqID in rIDs:
-      self.reqClient.deleteRequest(reqID)
+      self.reqClient.cancelRequest(reqID)
 
     return S_OK()
 
   def __removeWMSTasks(self, transJobIDs):
     """ wipe out jobs and their requests from the system
 
     TODO: should check request status, maybe FTS files as well ???
+    # 04.19: we now Cancel the request which is safer. FTS will notice that
+             and act accordingly.
```

Comment:
Should the TODO stay?

Reply:
At first I removed it, but then I wonder why was that todo ? Are there some cases where you do not want to cancel ? Canceling or removing does not really matter for the concerns of "checking before doing it".

Reply:
Well what's the difference between "cancel" and "delete"? Or maybe simply what does "cancel" mean?

Reply:
it's there : https://dirac.readthedocs.io/en/latest/AdministratorGuide/Systems/RequestManagement/rmsObjects.html?highlight=canceled#request :D 

When canceling a request, its status is just updated as "Canceled" in the DB, and will stay like this. If it is currently being processed by the FTSAgent or the REA, the ongoing execution will finish, but whenever they will put the Request in the DB, it will stay canceled, and no further actions will ever be taken on it. 

"delete" really removes the entry from the DB, which then drives the FTS3Agent crazy, or it might even pop back if the REA was working on it at this very moment

Reply:
Well then I think you can remove the "TODO".

Reply:
OK removed